server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java

```
@@ -9,30 +10,64 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Lazy;
+import org.springframework.core.task.AsyncTaskExecutor;
 import org.springframework.core.task.TaskDecorator;
+import org.springframework.core.task.TaskExecutor;
 import org.springframework.scheduling.annotation.AsyncAnnotationBeanPostProcessor;
 import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
-
-import static org.springframework.boot.autoconfigure.task.TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME;
+import org.springframework.web.servlet.config.annotation.AsyncSupportConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
 /**
  * Declares the registerSecurityServletFilter bean method to be processed by
  * the Spring container
  */
 @Configuration
 @EnableConfigurationProperties(PxfServerProperties.class)
-public class PxfConfiguration {
+public class PxfConfiguration implements WebMvcConfigurer {
+
+    /**
+     * Bean name of PXF's {@link TaskExecutor}.
+     */
+    public static final String PXF_APPLICATION_TASK_EXECUTOR_BEAN_NAME = "pxfApplicationTaskExecutor";
+
+    private final ListableBeanFactory beanFactory;
+
+    /**
+     * Constructs a PXF Configuration object with the provided
+     * {@link ListableBeanFactory}
+     *
+     * @param beanFactory the beanFactory
+     */
+    public PxfConfiguration(ListableBeanFactory beanFactory) {
+        this.beanFactory = beanFactory;
+    }
 
     /**
-     * Configures the TaskExecutorBuilder to be used for async requests (i.e. Bridge
+     * Configures the TaskExecutor to be used for async requests (i.e. Bridge
      * Read).
+     */
+    @Override
+    public void configureAsyncSupport(AsyncSupportConfigurer configurer) {
+        Object taskExecutor = this.beanFactory
+                .getBean(PXF_APPLICATION_TASK_EXECUTOR_BEAN_NAME);
+        if (taskExecutor instanceof AsyncTaskExecutor) {
```

why `if` ? We expect our own bean here, so cast or let it fail the startup if somehow misconfigured